### ISSUE TITLE
Survey mapping when you find where you are standing.

### ISSUE THREAD

**Issue by Tekkra** [dlebansais/PgSurveyor-Disclosed#2]
1. Craft a large number of surveys
2. Click through your surveys as normal to map them
3. Find one where you are standing
4. Do surveys
5. One survey will me marked on your map, but will not ever be pathed to
6. Run back to survey after - program crashes when you collect it

It appears when a survey is collected as you mapping them out that is likely tries to remove one from the list, thus ending up with n-1 surveys when it does the mapping calculations.  Then when you collect it after you walked off the end of the array and the program crashes.

Of course this is somewhat hard to replicate as you have to get a survey where you are standing, and that is very rare.

**Comment by dlebansais** [comment]
The tool supports collecting as you create maps, and if you don't click too fast you can see it makes room for the correct number of maps.
Now it should not crash, and it never does for me. I'll see if I can reproduce.

**Comment by Tekkra** [comment]
So it has been a bit, but I am not 100% sure what causes it, but I have seen it crash maybe a total of 2 times, and both of those times I gathered a survey where I was standing.  The 2nd time it just crashed immediately.  Talking to another person, they have seen the same thing once or twice as well.  Other than this it is rock solid.  Of course, a pain to reproduce.  Also, I have had it correctly update the maps as well when I find one where I stand, so it is unclear to me what I might have done different those times.

**Comment by Tekkra** [comment]
I figured it out, I believe I can fully replicate it!  I only saw it because of getting a map where you stood.  The problem actually is that I do on extra survey more than what is mapped.  When you get one where you stand, if you are not paying perfect attention and are clicking through them, you miss one on the middle, rather than remap them all, I just run those I have mapped and pin that last one and hit it after.  This is what causes it to crash.

I was able to map manually out 4 maps that were very close together.  I then started up the program and ran 3 of them, using a manual pin to know where that 4th one was.  I cleared the 3 as usual per the program and then ran over to the last one I had manually mapped, on gathering this map, the program crashed.

This being said if I run into this I can simply reset the program before getting that last map, and there will be no problems.
